pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +540,54 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+def _find_solve_with_eye(node):
+    valid_solves = (Solve, SolveTriangular)
+    # First, we look for the solve op
+    if not (
+        isinstance(node.op, Blockwise) and isinstance(node.op.core_op, valid_solves)
+    ):
+        return False
+    # Check whether second input to solve is Eye
+    solve_inputs = node.inputs
+    potential_eye_input = solve_inputs[1].owner
+    if not (potential_eye_input and isinstance(potential_eye_input.op, Eye)):
+        return False
+
+    return True
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_inv_inv(fgraph, node):
+    valid_inverses = (MatrixInverse, MatrixPinv, Solve, SolveTriangular)
+    valid_solves = (Solve, SolveTriangular)
```

Comment:
I know we always want to assume the input graph is valid, but we still might have to do some extra checks on `SolveTriangular` to make sure the `upper` argument is the same in both cases. Otherwise the graph is invalid anyway, but we'd be writing away a silent bug. Kind of a tricky situation?

We might also want to check that the assumptions on `Solve`, to make sure `assume_a` are the same for both. 

I strongly emphasize the use of the word  "might" in both of these suggestions, as I'm really not sure how much checking we should be responsible for in these types of rewrites.

Reply:
Checks we can do statically are totally fine, in those cases we usually don't rewrite and let it fail at runtime. But we could follow a global flag on what to do `ignore, warn, raise`, if the number of cases where we can tell something is wrong during rewrites (but not in make_node) grows.

Just Asserts that I have a problem with.

Reply:
Yeah I'm not suggesting we assert. With these solves though, `scipy` allows you to pass invalid inputs, and it happily returns the wrong result. You might remember this from #382. So if you have a non-triangular matrix (or a lower-triangular matrix with `upper=True`, `solve_triangular(solve_triangular(x, eye), eye)` will not be equal to `x`, although our rewrite will set it equal to `x` (which is perhaps what the user intended). The same is true for passing e.g. a non-symmetric matrix to `solve` with `assume_a='sym'`. 

I just wanted to flag this point, I'm not sure we should do anything about it, though.

Reply:
That's undefined behavior from scipy right? Not like exp(log(x)) where we keep nan if x<0. 

Here I think it's fine to assume it's valid because we couldn't reconstruct the invalid behavior trivially?